internal/keyvault/keyvault.go

```diff
@@ -0,0 +1,92 @@
+package keyvault
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"github.com/hashicorp/vault-client-go"
+	"github.com/hashicorp/vault-client-go/schema"
+	"github.com/notaryproject/notation-hashicorp-vault/internal/crypto"
+	"log"
+	"os"
+	"strings"
+	"time"
+)
+
+var VAULTADDR, VAULTTOKEN string
+
+var ParseCertificates = crypto.ParseCertificates
+
+type VaultClientWrapper struct {
+	vaultClient *vault.Client
+
+	keyID string
+}
+
+func NewVaultClientFromKeyID(id string) (*VaultClientWrapper, error) {
+	// read addr and token from environment variables
+	VAULTADDR = os.Getenv("VAULT_ADDR")
+	if len(VAULTADDR) < 1 {
+		log.Fatal("Error loading vault address")
+	}
+
+	VAULTTOKEN = os.Getenv("VAULT_TOKEN")
+	if len(VAULTTOKEN) < 1 {
+		log.Fatal("Error loading vault token")
+	}
+
+	// prepare a client with the given base address
+	client, err := vault.New(
+		vault.WithAddress(VAULTADDR),
+		vault.WithRequestTimeout(30*time.Second),
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// authenticate with a root token (insecure)
+	if err := client.SetToken(VAULTTOKEN); err != nil {
+		log.Fatal(err)
+	}
+
+	return &VaultClientWrapper{
+		vaultClient: client,
+		keyID:       id,
+	}, nil
+}
+
+func (vw *VaultClientWrapper) GetCertificateChain(ctx context.Context) ([]*x509.Certificate, error) {
+	// read a certChain
+	secret, err := vw.vaultClient.Secrets.KVv2Read(ctx, vw.keyID)
```

Comment:
@cipherboy It would be great if `Transit` could support uploading/downloading certificate chain in the future. For now, looks like we need to stick with the KVv2 option.